classes/proxy_lock_factory.php

```diff
@@ -285,9 +285,11 @@ private function log_lock($resourcekey) {
     private function log_unlock($resourcekey) {
         global $DB;
 
+        $resourcekeyprepared = $DB->get_manager()->generator->addslashes($resourcekey);
+
         $select = $DB->sql_compare_text('resourcekey') . ' = ' . $DB->sql_compare_text(':resourcekey');
 
-        $params = ['resourcekey' => $resourcekey];
+        $params = ['resourcekey' => $resourcekeyprepared];
```

Comment:
this line looks like it should not be changed.

I would try and make this patch only replace addslashes